nio-impl/src/main/java/org/xnio/nio/QueuedNioTcpServer.java

```diff
@@ -359,7 +359,9 @@ public void run() {
                         suspendedDueToWatermark = true;
                     }
                 } else if(suspendedDueToWatermark && openConnections <= getLowWater(connectionStatus)) {
-                    suspendedDueToWatermark = false;
+                    synchronized (QueuedNioTcpServer.this) {
```

Comment:
The field is volatile; is there a specific problem that adding synchronization solves?

Reply:
Ah, actually, there is no specific facing problem at this moment. I just noticed the different from the other parts which modify suspendedDueToWatermark. 

If synchronization is not required for this at all, I'm interested why do other parts have synchronized block? (Is the other synchronized blocks able to be eliminated, too?)

Anyway, this part is not main purpose of this PR. I'll remove this commit from this PR.